queries/bus.py

```diff
@@ -504,35 +518,41 @@ def query_arrival_time(query: Query, session: Session, result: Result):
                 if route is not None:
                     numbers.add(route.number)
                     stops_canonical.add(stop.name)
-        routes = sorted(list(numbers), key=lambda r: int(r))
-        if len(routes) != 1:
+
+        if len(numbers) != 1:
             # More than one route possible: ask user to clarify
-            route_seq = natlang_seq(list(map(str, routes)))
+            route_seq = natlang_seq(sorted(numbers, key=lambda n: int(n)))
             # "Einarsnesi eða Einarsnesi/Bauganesi"
-            stops_list = " eða ".join([to_dative(s) for s in stops_canonical])
-            answer = (
-                " ".join(["Leiðir", route_seq, "stoppa á", stops_list])
-                + ". Spurðu um eina þeirra."
+            stops_list = natlang_seq([dative_form(s) for s in sorted(stops_canonical)])
+            va_stops_list = " eða ".join(
+                [dative_form(s, voice=True) for s in sorted(stops_canonical)]
             )
+            answer = f"Leiðir {route_seq} stoppa á {stops_list}. Spurðu um eina þeirra."
```

Comment:
I know you didn't introduce this problem, and it is minor, but maybe worth adding as a TODO at the top of the document: Some bus stops will take the preposition "á" while others use "í". Ideally, this should be handled well ... we have some logic in geo.py if I remember correctly.

Reply:
Good point! Added the todo for the time being.